lib/org/openpsa/sales/sales/pdf.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * @package org.openpsa.sales
+ * @copyright CONTENT CONTROL http://www.contentcontrol-berlin.de/
+ * @license http://www.gnu.org/licenses/gpl.html GNU General Public License
+ */
+
+/**
+ * PDF Manager
+ *
+ * @package org.openpsa.sales
+ */
+class org_openpsa_sales_sales_pdf
+{
+    /**
+     * @var org_openpsa_sales_salesproject_dba
+     */
+    private $salesproject;
+
+    public function __construct(org_openpsa_sales_salesproject_dba $salesproject)
+    {
+        $this->salesproject = $salesproject;
+    }
+
+    /**
+     * @return midcom_db_attachment|NULL
+     */
+    public function get_attachment()
+    {
+        $pdf_files = org_openpsa_helpers::get_dm2_attachments($this->salesproject,"pdf_file");
+        if (!empty($pdf_files)) {
+            return reset($pdf_files);
+        }
+        return $this->render_and_attach();
+    }
+
+    public function get_button_options()
+    {
+        $message = "Are you sure to overwrite the current PDF-file?";
+
+        midcom\workflow\dialog::add_head_elements();
+        $l10n_midcom = midcom::get()->i18n->get_l10n();
+        $l10n = midcom::get()->i18n->get_l10n('org.openpsa.sales');
+
+        return array(
+            'data-dialog' => 'confirm',
+            'data-dialog-heading' => $l10n->get('create_pdf'),
+            'data-dialog-text' => $l10n->get($message),
+            'data-dialog-confirm-label' => $l10n_midcom->get('confirm'),
+            'data-dialog-cancel-label' => $l10n_midcom->get('cancel')
+        );
```

Comment:
This function also changed it's logic compared to the previous version: Now it alwas asks whether or not to overwrite the existing file, even if none exists